connectivity/curl/src/multi_performer.cpp

```diff
@@ -0,0 +1,131 @@
+// Copyright Antioch. All rights reserved.
+
+#include "multi_performer.h"
+
+#include <cstring>
+#include <iostream>
+#include <thread>
+
+#include "transfer.h"
+
+namespace antioch::connectivity::curl_transfer {
+
+MultiPerformer::MultiPerformer(CURLM* multi_handle) : multi_handle(multi_handle) {
+  std::thread thr(&MultiPerformer::run, this);
+  thr.detach();
+}
+
+MultiPerformer::~MultiPerformer() {
+  std::lock_guard<std::mutex> l(mtx);
+  std::cout << "got lock" << std::endl;
+  shutdown = true;
+  flag = true;
+  cv.notify_all();
+  for (const auto& pv : handle_objs) {
+    curl_multi_remove_handle(multi_handle, pv.first);
+  }
+  std::cout << "done dtor" << std::endl;
+}
+
+static size_t write_cb(void *data, size_t size, size_t nmemb, void *clientp) {
+  std::cout << "Called write_cb" << std::endl;
+  size_t realsize = size * nmemb;
+  auto mem = (EasyObj*)clientp;
```

Comment:
Use reinterpret_cast instead of c style cast

Reply:
since I want just a reinterpretation of memory, I'm gonna use a static cast. apparently it's best practice https://stackoverflow.com/questions/310451/should-i-use-static-cast-or-reinterpret-cast-when-casting-a-void-to-whatever